src/Microsoft.PowerShell.Commands.Utility/commands/utility/ConvertFromMarkdownCommand.cs

```diff
@@ -140,13 +141,43 @@ private void ConvertEachFile(IEnumerable<string> paths, MarkdownConversionType c
 
         private async Task<string> ReadContentFromFile(string filePath)
         {
-            Dbg.Diagnostics.Assert(File.Exists(filePath), "Caller should make sure the file exists.");
+            ErrorRecord errorRecord = null;
 
-            using (StreamReader reader = new StreamReader(new FileStream(filePath, FileMode.Open, FileAccess.Read, FileShare.Read)))
+            try
+            {
+                using (StreamReader reader = new StreamReader(new FileStream(filePath, FileMode.Open, FileAccess.Read, FileShare.Read)))
+                {
+                    string mdContent = await reader.ReadToEndAsync();
+                    return mdContent;
+                }
+            }
+            catch(FileNotFoundException fnfe)
```

Comment:
Maybe use one catch for all exceptions with custom (more useful) message and place original exception in InnerException?

Reply:
Since we are catching individual exceptions, I think the default messages are correct, without the need of custom messages. Examples below.

```
ConvertFrom-Markdown : Access to the path 'C:\Windows\System32\Configuration' is denied.
At line:1 char:1
+ ConvertFrom-Markdown -Path C:\Windows\System32\Configuration
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : SecurityError: (C:\Windows\System32\Configuration:String) [ConvertFrom-Markdown], UnauthorizedAccessException
+ FullyQualifiedErrorId : FileUnauthorizedAccess,Microsoft.PowerShell.Commands.ConvertFromMarkdownCommand
```

```
ConvertFrom-Markdown : Cannot find path 'D:\temp\sdfsdfs.md' because it does not exist.
At line:1 char:1
+ ConvertFrom-Markdown -Path D:\temp\\sdfsdfs.md
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : ResourceUnavailable: (D:\temp\\sdfsdfs.md:String) [ConvertFrom-Markdown], ItemNotFoundException
+ FullyQualifiedErrorId : FileNotFound,Microsoft.PowerShell.Commands.ConvertFromMarkdownCommand
```